packages/puppeteer-core/src/bidi/core/BrowsingContext.ts

```diff
@@ -464,6 +472,39 @@ export class BrowsingContext extends EventEmitter<{
     await this.userContext.browser.removePreloadScript(script);
   }
 
+  @throwIfDisposed<BrowsingContext>(context => {
+    // SAFETY: Disposal implies this exists.
+    return context.#reason!;
+  })
+  async getCookies(
+    options: GetCookiesOptions = {}
+  ): Promise<Bidi.Network.Cookie[]> {
+    const {
+      result: {cookies},
+    } = await this.#session.send('storage.getCookies', {
```

Comment:
let's have a single place doing calls and these methods should delete to UserContext.* which should accept an optional BrowsingContext as a parameter?

Reply:
That would break encapsulation. You could pass in the browsing context from a different user context which would pass but technically not be a valid operation.

Reply:
One could check if the context belongs, right? `if (this.id !== browsingContext.id) throw`.

Reply:
Sure, the check can happen, but why check something when the operation is independent of the user context

Reply:
In the spec, the browsing context is only a shortcut to determine the user context + source origin. So to match the spec structure, the client should be also delegating to the user context: otherwise, if new functionality is added that applied to both kinds of keys, the client needs to be modified in two places and the implementation has a potential to diverge.